### PR TITLE
Improve sync

### PR DESCRIPTION
**Pull request by HansOlsson**
This is intended to clarify handling of some odd cases - especially for the clocked discretized partition.
Each commit is explained in some detail.

There are a number of additional issues that are not part of this PR, but will be linked.

### CODE REVIEW

File: chapters/synchronous.tex [hunk 1]
```diff
@@ -1050,9 +1057,11 @@ \section{Continuous-Time Equations in Clocked Partitions}\label{continuous-time-
 \end{nonnormative}
 
 \begin{nonnormative}
-It is not defined, how events are handled that appear when solving the continuous-time partition. For example, a tool could handle events exactly in the same way as for a usual
-simulation.  Alternatively, relations might be interpreted literally, so that events are no longer triggered (in order that the time for an integration step is always the same,
-as needed for hard real-time requirements).
+It is not defined, how events are handled that appear when solving the continuous-time partition.
```
Comment (henrikt-ma) [inline chapters/synchronous.tex:1060]: Maybe rephrase a little?
```suggestion
It is not defined how to handle events that are triggered while solving the continuous-time partition.
```

File: chapters/synchronous.tex [hunk 2]
```diff
@@ -1050,9 +1057,11 @@ \section{Continuous-Time Equations in Clocked Partitions}\label{continuous-time-
...
+For example, a tool could handle events in the same way as for a usual simulation - but only check them at the time associated with clock-ticks.
```
Comment (henrikt-ma) [inline chapters/synchronous.tex:1061]: En-dash:
```suggestion
For example, a tool could handle events in the same way as for a usual simulation -- but only check them at the time associated with clock-ticks.
```

File: chapters/synchronous.tex [hunk 3]
```diff
@@ -883,6 +883,13 @@ \subsection{Connected Components of the Equations and Variables Graph}\label{con
 \emph{Connected components}) of this graph with appropriate definition of
 the incidence operator.
 
+An special case is the built-in variable \lstinline!time! (see \cref{built-in-variable-time}).
+Each use of \lstinline!time! is conceptually included as a separate variable in this analysis, \lstinline!$\text{time}_i$! with \lstinline!der($\text{time}_i$)=1!.
```
Comment (henrikt-ma) [inline chapters/synchronous.tex:887]: Formatting:
```suggestion
Each use of \lstinline!time! is conceptually included as a separate variable in this analysis, $\mathit{time}_i$ with $\text{\lstinline!der!}(\mathit{time}_{i}) = 1$.
```

**Review by eshmoylova — APPROVED** [review]

File: chapters/synchronous.tex [hunk 4]
```diff
@@ -1083,6 +1092,13 @@ \subsection{Clocked Discrete-Time and Clocked Discretized Continuous-Time Partit
 Such a partition has to be solved with a \emph{solver method} of \cref{solver-methods}.
 When \lstinline!previous(x)! is used on a continuous-time state variable \lstinline!x!, then \lstinline!previous(x)! uses the start value of \lstinline!x! as value for the first clock tick.
 
+The use of the operator sample from \cref{event-related-operators-with-function-syntax} in a clocked partition is problematic.
```
Comment (henrikt-ma) [inline chapters/synchronous.tex:1095]: Formatting:
```suggestion
The use of the operator \lstinline!sample! from \cref{event-related-operators-with-function-syntax} in a clocked partition is problematic.
```

File: chapters/synchronous.tex [hunk 5]
```diff
@@ -1083,6 +1092,13 @@ \subsection{Clocked Discrete-Time and Clocked Discretized Continuous-Time Partit
...
+Diagnostics are appropriate especially if it is intended to generate sampling faster than the clocks, and otherwise the sampling should ideally be adjusted to the clock ticks.
+\begin{nonnormative}
```
Comment (henrikt-ma) [inline chapters/synchronous.tex:1097]: Empty lines on both sides of `nonnormative` increase readability of the source:
```suggestion

\begin{nonnormative}
```

File: chapters/synchronous.tex [hunk 6]
```diff
@@ -1083,6 +1092,13 @@ \subsection{Clocked Discrete-Time and Clocked Discretized Continuous-Time Partit
...
+Since the goal is that it should be possible to use \emph{every} continuous-time Modelica model, it is not forbidden.
```
Comment (henrikt-ma) [inline chapters/synchronous.tex:1098]: It may not be obvious what kind of use this refers to;
```suggestion
Since the goal is that it should be possible to use \emph{every} continuous-time Modelica model synchronously, it is not forbidden.
```

I'm afraid that this way of using _synchronously_ could be new, and maybe we should try to stick with some other formation that is already in use?

File: chapters/synchronous.tex [hunk 7]
```diff
@@ -1083,6 +1092,13 @@ \subsection{Clocked Discrete-Time and Clocked Discretized Continuous-Time Partit
 Such a partition has to be solved with a \emph{solver method} of \cref{solver-methods}.
 When \lstinline!previous(x)! is used on a continuous-time state variable \lstinline!x!, then \lstinline!previous(x)! uses the start value of \lstinline!x! as value for the first clock tick.
 
+The use of the operator sample from \cref{event-related-operators-with-function-syntax} in a clocked partition is problematic.
+Diagnostics are appropriate especially if it is intended to generate sampling faster than the clocks, and otherwise the sampling should ideally be adjusted to the clock ticks.
```
Comment (henrikt-ma) [inline chapters/synchronous.tex:1096]: To me it seems that this is a normative change that should require a language group decision, as the previous state didn't make clear that the problematic `sample` really is allowed.

**Review by henrikt-ma — CHANGES_REQUESTED** [review]
Marking as _Request changes_, mainly to serve as reminder that a language group decision seems desirable before accepting these changes.

File: chapters/synchronous.tex (hunk 7 above)
Comment (HansOlsson) [reply chapters/synchronous.tex:1096]: I can separate out that to another PR, to have a clearer base for discussion.

Comment (henrikt-ma) [reply chapters/synchronous.tex:1096]: Sounds like a good idea.

File: chapters/synchronous.tex (hunk 6 above)
Comment (HansOlsson) [reply chapters/synchronous.tex]: Reused 'sampled data control system' from section introduction.

File: chapters/synchronous.tex (hunk 7 above)
Comment (HansOlsson) [reply chapters/synchronous.tex]: This has now been done.

Comment (HansOlsson) [reply chapters/synchronous.tex]: Which makes this PR unproblematic.

File: chapters/synchronous.tex (hunk 5 above)
Comment (henrikt-ma) [reply chapters/synchronous.tex]: Neither resolved nor commented.  Marked as resolved by mistake?

File: chapters/synchronous.tex (hunk 6 above)
Comment (henrikt-ma) [reply chapters/synchronous.tex]: Obsolete after breaking out this part as a separate PR.  Resolving.

File: chapters/synchronous.tex (hunk 7 above)
Comment (henrikt-ma) [reply chapters/synchronous.tex]: Moved to #3099.  Resolving.

File: chapters/synchronous.tex [hunk 8]
```diff
@@ -883,6 +883,13 @@ \subsection{Connected Components of the Equations and Variables Graph}\label{con
 \emph{Connected components}) of this graph with appropriate definition of
 the incidence operator.
 
+A special case is the built-in variable \lstinline!time! (see \cref{built-in-variable-time}).
+Each use of \lstinline!time! is conceptually included as a separate variable in this analysis, $\mathit{time}_i$ with $\text{\lstinline!der!}(\mathit{time}_{i}) = 1$.
```
Comment (henrikt-ma) [inline chapters/synchronous.tex:887]: Is it obvious that this is a trivial change?  Couldn't `time` have been defined using `interval()` instead to avoid the need for discretization?  Just asking.

**Review by henrikt-ma — COMMENTED** [review]
It would be good if an expert on synchronous language features could review the suggested handling of `time` – see comment.

**Review by henrikt-ma — DISMISSED** [review]

**Review by henrikt-ma — COMMENTED** [review]
It would be good if an expert on synchronous language features could review the suggested handling of `time` – see comment.

File: chapters/synchronous.tex (hunk 5 above)
Comment (HansOlsson) [reply chapters/synchronous.tex]: No, I believe it was merged, **and then** that entire sample-part was cut out to a separate PR #3099 - where you can see the empty lines.

File: chapters/synchronous.tex (hunk 8 above)
Comment (HansOlsson) [reply chapters/synchronous.tex:887]: There are always alternatives.

However, as I understand it: forbidding direct use of time for the non-discretized variant was part of the design goal (since expressions such as `time>=0.2` have issues with numerical noise that we want to avoid), and you can see that examples use `sample(time)`; and clearly that is one option - but that clashes too much with having every continuous model in a sampled systems. Thus the above keeps the previous intentions.

File: chapters/synchronous.tex (hunk 5 above)
Comment (henrikt-ma) [reply chapters/synchronous.tex]: OK, resolving.

File: chapters/synchronous.tex (hunk 8 above)
Comment (henrikt-ma) [reply chapters/synchronous.tex:887]: I suppose it comes down to whether you want the use of `time` to imply the need for time discretization and a `solverMethod`, or if you want to be able to use `time` in a clocked discrete-time partition?

Comment (HansOlsson) [reply chapters/synchronous.tex:887]: Yes. And it is clear that the original design was to not allow `time `directly in clocked discrete-time partitions (to offer stronger guarantees for synchronous), and this keeps that design - while still allowing `time `with `solverMethod`, since it is not worse than a differential equation.

Comment (HansOlsson) [reply chapters/synchronous.tex:887]: @bernhard-thiele @MartinOtter Do you have any input on this?

**Comment by HansOlsson** [comment]
(Comments from synchronous experts still appreciated.)
Creating a normal vote for this (will send mail as well) - use thumbs up (accept) or down (against) - under the face to the right.

File: chapters/synchronous.tex (hunk 8 above)
Comment (HansOlsson) [reply chapters/synchronous.tex:887]: 11 days. I guess not
